packages/esm-service-queues-app/src/hooks/useConcept.ts

```diff
@@ -0,0 +1,13 @@
+import { openmrsFetch, restBaseUrl } from '@openmrs/esm-framework';
+import useSWRImmutable from 'swr/immutable';
+import { type Concept } from '../types';
+
+export function useConcept(uuid: string) {
```

Comment:
I was wondering if such common/shared functions can be consolidated in a shared lib for reuse. Many of us will keep defining such in every app that interact with concepts. In the absence of such, we have to go this route

Reply:
@ibacher what is the best practice?

Reply:
Well, I'd probably think that, for things like this, we'd have `useConceptSet()` or something. In general, my biggest concern is that loading one concept at a time is one HTTP request per concept, so usually we'd want to bulk load a series of concepts, such as using the `/conceptreferences` endpoint.

Reply:
That makes sense to me. I know one challenge we've had is the need to edit our dictionaries to take care of the concept set and the members. Some implementers use the dictionary as it comes from i.e. OCL exports. The guidance is clear though. Thanks

Reply:
These aren't mutually exclusive, and in this case - loading a Concept and loading a ConceptSet are exactly the same thing, as a ConceptSet is just a Concept that has setMembers.  So it isn't really an anti-pattern in this case, there is exactly one request.

I 1000% think that we should have things like "useConcept" and "useSystemSetting" already defined and in esm-core somewhere (eg. esm-api), and I am continually shocked that we don't.  It cost me a non-zero amount of time to create these, and I don't want to have to independently create (and test!) all of these everywhere we want to use them.

@ojwanganto - I fully intend to endorse moving things to esm-api.  I just didn't want any philosophical debates muddying this PR, or any dependencies on getting this merged in.  Let's ticket and do that separately.